server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Requeue.java

```diff
@@ -0,0 +1,87 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.mail.MessagingException;
+
+import org.apache.james.lifecycle.api.LifecycleUtil;
+import org.apache.james.queue.api.MailQueue;
+import org.apache.james.queue.api.MailQueueFactory;
+import org.apache.james.queue.api.MailQueueName;
+import org.apache.james.util.DurationParser;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMailet;
+
+import com.google.common.base.Preconditions;
+
+public class Requeue extends GenericMailet {
+    private final MailQueueFactory<?> mailQueueFactory;
+
+    private MailQueue mailQueue;
+    private Optional<Duration> delayDuration;
+    private String processor;
+    private String consumeProcessor;
+
+    @Inject
+    public Requeue(MailQueueFactory<?> mailQueueFactory) {
+        this.mailQueueFactory = mailQueueFactory;
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        MailQueueName mailQueueName = Optional.ofNullable(getInitParameter("queue"))
+            .map(MailQueueName::of).orElse(MailQueueFactory.SPOOL);
+
+        mailQueue = mailQueueFactory.getQueue(mailQueueName)
+            .orElseThrow(() -> new IllegalArgumentException("Can not find queue " + mailQueueName.asString()));
+        delayDuration = Optional.ofNullable(getInitParameter("delay"))
+            .map(delayValue -> DurationParser.parse(delayValue, ChronoUnit.SECONDS));
+        processor = Optional.ofNullable(getInitParameter("processor"))
+            .orElse(Mail.DEFAULT);
+        consumeProcessor = Optional.ofNullable(getInitParameter("consume"))
+            .orElse(Mail.GHOST);
```

Comment:
If we consume the mail, isn't it always GHOST anyway? why need a param here? I saw it was one of @chibenwa comment but I'm not sure to get it :)

Reply:
If we don't have a consume for handler current mail, The mail can be sent duplicate. 
IMO, because we re-queue new mail after delay, we should stop the process of current mail, Ghost for that

Reply:
> If we consume the mail, isn't it always GHOST anyway?

Yes it is

Reply:
So what do we want here? to have the possibility to consume the mail? Isn't a boolean consume true/false making more sense then if to consume we use ghost? or there could be a usage issue with a wrong configuration?